File: TransCoder_tokenized_test_set_functions/cpp/REPLACE_OCCURRENCES_STRING_AB_C_WITHOUT_USING_EXTRA_SPACE.cpp
void translate ( char * str ) {
  if ( str [ 0 ] == '' ) return;
  for ( int i = 1;
  str [ i ] != '';
  i ++ ) {
    if ( str [ i - 1 ] == 'A' && str [ i ] == 'B' ) {
      str [ i - 1 ] = 'C';
      for ( int j = i;
      str [ j ] != '';
      j ++ ) str [ j ] = str [ j + 1 ];
    }
  }
  return;
}

File: TransCoder_tokenized_test_set_functions/java/REPLACE_OCCURRENCES_STRING_AB_C_WITHOUT_USING_EXTRA_SPACE.java
static void translate ( char str [ ] ) {
  for ( int i = 1;
  i < str . length;
  i ++ ) {
    if ( str [ i - 1 ] == 'A' && str [ i ] == 'B' ) {
      str [ i - 1 ] = 'C';
      int j;
      for ( j = i;
      j < str . length - 1;
      j ++ ) str [ j ] = str [ j + 1 ];
      str [ j ] = ' ';
    }
  }
  return;
}